src/charm.py

```diff
@@ -130,12 +128,10 @@ def state(self) -> State | None:
             raise exc
         except IncompleteStateError as exc:
             logger.debug("Charm configuration not ready, %s", exc)
-            self.unit.status = ops.WaitingStatus("Charm state is not yet ready")
-            return None
+            raise exc
```

Comment:
I think it'd be simpler to remove all these `except` clauses and log the exception in the reconcile loop instead